### PR TITLE
TOOLS/PERF: Report error when using non-host memory type for UCP RMA and AMO

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Report error when using non-host memory type for UCP RMA and AMO

## Why ?

We should report error to user instead of crashing/failing or etc in UCP level

## How ?

1. Add new check for API type, memory type and command in ucx_perf_test_check_params
2. Print error only in case of it was requested by flags

### CODE REVIEW

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 10-Jan-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/43d21df75c1aa3be053790c37eada83973ee56dd/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/207083b5cc7fe713a53138a39a45d40c44507f7a/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/dc863c87fb014a97eb1963342ae610397e9c7543/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/95dbecb5202078153f731cc0bf12f7a7d3355b40/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/56116047a7b9fb46b07a9cb5ba9f54468a1e8cf7/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/dd10a3c876c6336edcb019111015f70bd3104bd6/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/9097105b41e49c6c65b515781509a47386284c38/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/476f9655490176055a541d760ff957ed397533aa/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/5734f263e0e927552305566cbd3e0797d5da252d/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/08f82c0c091a3a1483590776524b56eef806892e/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/dfeca83c7d44f0720beac8ce35be2e48604f2289/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/8f9fe5a01695c4d8b5c145ab3b3949e43a5927ca/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/d2743f37c2d5b0b4a691274785478b5ed7625b5b/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/5c3dddbef4e12024317b2aec648d0cb8aeb957ee/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/2e5243748aa299e68a58e5cc0c86abdfb572d601/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/8a2cc16b0076b35f58502277c418847ee6bf1fbf/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/259ce7f47184073b947ce968a9a06245703ae84d/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/150723632ea660a19e1e5a5e56576d86f68e7a54/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/d7a4dad5e387f2c68ba137933db3b9948c438091/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/3fa1457a44488c94472c7850e25029feb1b0e560/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/7cfd928de4ab0f5e2fc13b6853929e67d44fc52f/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/70ad06e1a1f8187c8124cea30a71c379ebeb18f0/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/60f44724d3fce3ee1e4934923c2b21a4ccd93fce/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/2594307986c7e6e5082cc0f0946ad0515bc799c1/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/52fa7bd8763194d1ce434e4436f3885c/raw/632177a9e18ec9872aec724b10d8c1f7efa87a7d/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by dmitrygx** [comment]
bot:pipe:retest

**Comment by dmitrygx** [comment]
@yosefe could you review pls?

**Comment by brminich** [comment]
@alinask, can you plz review?

**Comment by alinask** [comment]
@dmitrygx note the conflict please.

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 16-Jan-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/c6035bfbf1dd41d23bb87a57fb81132c64617929/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/96f4277cff898dde10cffea46e2bacc8ed1ecf85/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/e495826e683525a68fb96be0c39a21ad6f49d5bd/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/263fddf30b072634d17b82ec0a453d128782acdf/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/f42d9f4ca3a628d2ca35e82ef89216aef74e4b2b/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/345ef41a0f6c791daae030f1465e9ee63f21d130/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/aff9b7d09553b731c39a2a439ab58618f2c3d807/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/f5d72eba5bc7f16b664041b581bdfd1583bc77a0/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/434d3a0b04b9715f816106ddcdf3db49a799ca45/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/2e4d164f22c4261ec9b3167503d4ba01f0d96ea9/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/ea8a4bcc0a00f191f9910aae538c24ef1cb6d803/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/27b73a24f9f83e458afef4df1398109a47693908/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/352801b1abaead961f703937336ba2fd39018442/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/1420c30c3375b5591e0b7b50e5794e59afe97656/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/a3c0339bc865871c654502aac8be2475fe93d0fa/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/fced9d4aff68f86a4af5d847db8ebcb7d767dbb4/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/d887ebeaee4b93697e4d7c8aa9b13e99a72ff311/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/beafcdcecb4f8690cd48cd04af553fc90602558e/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/9fc5fd25a3cfa0e28287c27418e8ee2a1b3e359f/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/7dca750488dfcbfff959437a890c598429678e3c/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/7a60caa3784ebd42ebbf4ed2b88a98e4dfd2b299/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/c638d428b89b15b00540758dfdaf24f5bd5c9888/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/0642a5d7e3fa02b399ca514868ef35adfe00e07e/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/1bf0fcf72b28fd02899e8a603c69d2d3fb5169f6/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/eb4f596ed5793f798c2506c694fe4231/raw/8454e8036dd57cf24e9487a771f33429ed1e5220/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Comment by dmitrygx** [comment]
@alinask @brminich ok to merge?

File: src/tools/perf/lib/libperf.c [hunk 1]
```diff
@@ -678,7 +698,7 @@ static ucs_status_t uct_perf_test_check_capabilities(ucx_perf_params_t *params,
     status = uct_perf_test_check_md_support(params, params->recv_mem_type, &md_attr);
     if (status != UCS_OK) {
         return status;
-    }
+    }
```
Comment (brminich) [inline src/tools/perf/lib/libperf.c]: sorry, just noticed extra space after `} `
please fix it

Comment (dmitrygx) [reply src/tools/perf/lib/libperf.c]: @brminich good catch
fixed

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 18-Jan-2020

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/d5eca96184a4774c7e0a3e8d25ca4a3ceac3bac4/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/bfe0a87ec1044cb9e13e09f7d0d49faaacd94e27/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/92bc4d24bc975264ef09c7cf84311141742c5493/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/6cf8c3af196676ce8cc90124968ec972f3b62b75/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/d0a02257607afa5a40ed53169d59a91b25973c4c/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/94fb49fbb9671f5ee8f575c50e68b831e58478e9/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/d6b4ef8aaf98aa59a5155ac958373843a385a859/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/78282f9b14d22716afccdbdee9967d7db09eb6e1/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/7f362d1c2b563ede9d7f3775ca7f2ba0b8064df9/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/666e4156868fe3ece2f2ad4cfba9dc882487f834/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/849ee0568977dd2f7369dcc75818cc2024178672/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/ede1e75e31e0c9b12951c6f6b8799b6f1d9a7a48/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/e3f2d037e2a5a2e7e5a22ed4e93411d9fd9aa781/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/d4f454591e0c2d28a33d788708d992e89b795643/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/3db4cc76d83519be1a4bf7ecd98823e760c2665e/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/93236d091dcbe18aa9b37ac586a7fb17afecb6db/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/b53064a91f1c6c3c7659125a32d24eed7a0b479b/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/cc08be748718e482273d0e9c3bc5b7b74f2c4b4d/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/9ad63c2860fe95160b253b36edf4783d44714160/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/11922dca5dc00888d96c9d1b7f4a5c1c61011f4d/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/f7dd01f8a60d80506d88ec11ff3b1c53c7f0a123/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/78f10170db95d2e2fd6bd8df9550750c4db5fcf0/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/3255229abf10efb6ffea73889e3171cb2757d9b6/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/3e03936514ecf58d675945aae2a107c1a79e6e1f/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/a478203aa5da30bc3417dfeb4f200d49/raw/2c4cf18b1321a9ee061003e1e4003012113a9d3c/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**